src/layer.rs

```diff
@@ -144,11 +152,22 @@ impl LayerSet {
             Err(NamingError::Duplicate(name.to_string()))
         } else {
             let name = Name::new(name).map_err(|_| NamingError::Invalid(name.into()))?;
-            let path = crate::util::default_file_name_for_layer_name(&name, &self.path_set);
-            let layer = Layer::new(name, path);
-            self.path_set.insert(layer.path.to_string_lossy().to_lowercase());
-            self.layers.push(layer);
-            Ok(self.layers.last_mut().unwrap())
+            Ok(self.create_layer(name))
+        }
+    }
+
+    /// Returns a mutable reference to a layer, by name, creating the layer if it doesn't exist
+    pub fn get_or_create_layer(&mut self, name: &str) -> Result<&mut Layer, NamingError> {
+        let index = self.layers.iter().position(|l| l.name.as_str() == name);
+        match index {
+            Some(its_here) => Ok(&mut self.layers[its_here]),
+            None => {
+                // We can avoid the formal checks that new_layer does because by checking if it's already in the list,
```

Comment:
I think we still want to check that we aren't `DEFAULT_LAYER_NAME`, since it's possible that the default layer has another name.

See this test case that was deleted when we removed the previous version:

```rust
#[test]
     #[should_panic(expected = "reserved")]
     fn get_or_create_false_default_layer() {
         let mut layer_set = LayerSet::default();
         layer_set.rename_layer("public.default", "foreground", false).unwrap();
         layer_set.get_or_create("public.default");
     }
```


and... I think honestly it's easiest/hardest to screw up if t his *does* just call the old `new_layer` method if we find nothing. In practice this isn't called much, and iterating even small-10s of layers is still going to be very fast, and then we can be confident we're always using the same logic.